security/cmd/flexvolume/driver/driver.go

```diff
@@ -0,0 +1,515 @@
+			Serviceaccount: ninputs.ServiceAccount,
+		},
+		Workloadpath: ninputs.UID,
+	}
+	return &wlInfo, true
+}
+
+// doMount handles a new workload mounting the flex volume drive. It will:
+// * mount a tmpfs at the destination directory of the workload created by the kubelet.
+// * create a sub-directory ('nodeagent') there
+// * do a bind mount of the nodeagent's directory on the node to the destinationDir/nodeagent.
+func doMount(destinationDir string, ninputs *pb.WorkloadInfo) error {
+	newDir := filepath.Join(configuration.NodeAgentWorkloadHomeDir, ninputs.Workloadpath)
+	err := os.MkdirAll(newDir, 0777)
+	if err != nil {
+		return err
+	}
+
+	// Not really needed but attempt to workaround:
+	// https://github.com/kubernetes/kubernetes/blob/61ac9d46382884a8bd9e228da22bca5817f6d226/pkg/util/mount/mount_linux.go
+	cmdMount := getExecCmd("/bin/mount", "-t", "tmpfs", "-o", "size=8K", "tmpfs", destinationDir)
+	err = cmdMount.Run()
+	if err != nil {
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	newDestinationDir := filepath.Join(destinationDir, "nodeagent")
+	err = os.MkdirAll(newDestinationDir, 0777)
+	if err != nil {
+		cmd := getExecCmd("/bin/unmount", destinationDir)
+		cmd.Run()            //nolint: errcheck
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	// Do a bind mount
+	cmd := getExecCmd("/bin/mount", "--bind", newDir, newDestinationDir)
+	err = cmd.Run()
+	if err != nil {
+		cmd = getExecCmd("/bin/umount", destinationDir)
+		cmd.Run()            //nolint: errcheck
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	return nil
+}
+
+// doUnmount will unmount the directory
+func doUnmount(dir string) error {
+	cmd := getExecCmd("/bin/umount", dir)
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Rollback the operations done by doMount
+// rollback code is best effort attempts to clean up.
+// nolint: errcheck
+func handleErrMount(destinationDir string, ninputs *pb.WorkloadInfo) {
+	newDestinationDir := filepath.Join(destinationDir, "nodeagent")
+	doUnmount(newDestinationDir)
+	doUnmount(destinationDir)
+	// Sufficient to just remove the underlying directory.
+	os.RemoveAll(destinationDir)
+	os.RemoveAll(filepath.Join(configuration.NodeAgentWorkloadHomeDir, ninputs.Workloadpath))
+}
+
+// Mount handles the mount command to the driver.
+func Mount(dir, opts string) error {
+	inp := strings.Join([]string{dir, opts}, "|")
+
+	ninputs, s := checkValidMountOpts(opts)
+	if s == false {
+		return failure("mount", inp, "Incomplete inputs")
+	}
+
+	if err := doMount(dir, ninputs); err != nil {
+		sErr := "Failure to mount: " + err.Error()
+		return failure("mount", inp, sErr)
+	}
+
+	if configuration.UseGrpc == true {
+		if err := sendWorkloadAdded(ninputs); err != nil {
+			handleErrMount(dir, ninputs)
+			sErr := "Failure to notify nodeagent: " + err.Error()
+			return failure("mount", inp, sErr)
+		}
+	} else {
+		if err := addCredentialFile(ninputs); err != nil {
```

Comment:
i think we can combine this else with previous if.

Reply:
@wattli Thanks for the review. I've pushed a commit to address it.